api/expiring_license_service.go

```diff
@@ -58,26 +57,28 @@ func (a Api) ListExpiringLicenses(expiresWithin string, staged bool, deployed bo
 	return expiredLicense, err
 }
 
-func (a Api) getProductsLicenseInfo(expiringProducts *[]expiringProduct, staged bool, deployed bool) error {
+func (a Api) getProductsLicenseInfo(staged bool, deployed bool) ([]expiringProduct, error) {
+	expiringProducts := []expiringProduct{}
```

Comment:
i think we should probably not pass the expiringProducts as reference here too, and call `addStagedProducts` as `getStagedProducts` and `addDeployedProducts` as `getDeployedProducts` and merge the slices from both the methods and remove the duplicates if any